src/test/java/org/spongepowered/api/command/args/parsing/SpaceSplitInputTokenizerTest.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.command.args.parsing;
+
+import static org.junit.Assert.assertEquals;
+
+import com.google.common.collect.ImmutableList;
+import org.junit.Test;
+import org.spongepowered.api.command.args.ArgumentParseException;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class SpaceSplitInputTokenizerTest {
+
+    private static List<String> parseFrom(String args) throws ArgumentParseException {
+        return InputTokenizer.spaceSplitString().tokenize(args, false).stream().map(SingleArg::getValue).collect(Collectors.toList());
+    }
+
+    @Test
+    public void testEmptyString() throws ArgumentParseException {
+        assertEquals(ImmutableList.of(), parseFrom(""));
+    }
+
+    @Test
+    public void testSpaceString() throws ArgumentParseException {
+        assertEquals(ImmutableList.of(""), parseFrom(" "));
```

Comment:
I suspect I missed this test now I look at this, if that's the case, feel free to update _my_ changes to accomodate this test (and include this test in the space tokenizer tests). It should be as simple as removing the regex check, I thought a string of spaces was no arguments.

That said, I'm not convinced that we should return any arguments at all if we have just a load of spaces, but the other side of this is that trailing spaces do create an extra arg at the end.

Dunno if anyone else has an opinion on that.

Reply:
This is consistent with the behavior of `QuotedStringsTokenzier` at the moment. I *believe* the fringe case is where you have something like `"/command  "` (double space), where that results in a leading space at the start of the argument (split doesn't remove this as it uses a negative index to separate the command from the argument). I'd have to dig through to check if this theory is correct, but for the meantime it should be reasonable enough.